crates/ruff_linter/resources/test/fixtures/flake8_pytest_style/PT017.py

```diff
@@ -12,6 +12,14 @@ def test_ok():
     assert e.value.message
 
 
+def allow_exception_in_message():
+    x = 0
+    try:
+        1 / x
+    except ZeroDivisionError as e:
+        assert x == 0, e
```

Comment:
To me this feels like exactly the kind of thing the rule is trying to catch, rather than something we should avoid emitting a diagnostic for: I commented at https://github.com/astral-sh/ruff/issues/11869#issuecomment-2168117376. If you want to use the value of the exception in the assertion message, you can do that using `pytest.raises()`:

```py
def allow_exception_in_message():
    x = 0
    with pytest.raises(ZeroDivisionError) as exc_info:
        1 / x
    assert x == 0, exc_info.value
```

Reply:
I'm leaning towards agreeing with Alex here